File: 1сем/ОАиП лабы/Common7/IDE/ProjectTemplatesCache/VisualBasic/WCF/1049/RssService/Feed1.vb
' ПРИМЕЧАНИЕ. Можно использовать команду "Переименовать" в контекстном меню, чтобы изменить имя класса "Feed1" в коде и файле конфигурации.
Public Class Feed1
    Implements IFeed1

    Public Function CreateFeed() As SyndicationFeedFormatter Implements IFeed1.CreateFeed
        ' Создать новый веб-канал.
        Dim feed As New SyndicationFeed("Feed Title", "A WCF Syndication Feed", Nothing)
        Dim items As New List(Of SyndicationItem)

        ' Создать новый элемент рассылки.
        Dim item As New SyndicationItem("An item", "Item content", Nothing)
        items.Add(item)
        feed.Items = items

        ' Возвращать канал ATOM или RSS, основываясь на строке запроса
        ' RSS-&gt; http://localhost:8733/Design_Time_Addresses/$safeprojectname$/Feed1/
        ' Atom-&gt; http://localhost:8733/Design_Time_Addresses/$safeprojectname$/Feed1/?format=atom
        Dim query As String = WebOperationContext.Current.IncomingRequest.UriTemplateMatch.QueryParameters.Get("format")
        Dim formatter As SyndicationFeedFormatter = Nothing
        If (query = "atom") Then
            formatter = New Atom10FeedFormatter(feed)
        Else
            formatter = New Rss20FeedFormatter(feed)
        End If

        Return formatter
    End Function

End Class

File: 1сем/ОАиП лабы/Common7/IDE/ProjectTemplatesCache/CSharp/WCF/1049/RssService/Feed1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.ServiceModel.Syndication;
using System.ServiceModel.Web;
using System.Text;

namespace $safeprojectname$
{
	// ПРИМЕЧАНИЕ. Команду "Переименовать" в меню "Рефакторинг" можно использовать для одновременного изменения имени класса "Feed1" в коде и файле конфигурации.
	public class Feed1 : IFeed1
	{
		public SyndicationFeedFormatter CreateFeed()
		{
			// Создать новый веб-канал.
			SyndicationFeed feed = new SyndicationFeed("Feed Title", "A WCF Syndication Feed", null);
			List<SyndicationItem> items = new List<SyndicationItem>();

			// Создать новый элемент рассылки.
			SyndicationItem item = new SyndicationItem("An item", "Item content", null);
			items.Add(item);
			feed.Items = items;

			// Возвращать канал ATOM или RSS, основываясь на строке запроса
			// RSS-&gt; http://localhost:8733/Design_Time_Addresses/$safeprojectname$/Feed1/
			// Atom-&gt; http://localhost:8733/Design_Time_Addresses/$safeprojectname$/Feed1/?format=atom
			string query = WebOperationContext.Current.IncomingRequest.UriTemplateMatch.QueryParameters["format"];
			SyndicationFeedFormatter formatter = null;
			if (query == "atom")
			{
				formatter = new Atom10FeedFormatter(feed);
			}
			else
			{
				formatter = new Rss20FeedFormatter(feed);
			}

			return formatter;
		}
	}
}
